File: Data Structures & Algorithms/two-integer-sum-ii/submission-2.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        # i , j = 0 , len(numbers) - 1
        i = 0
        index = 0
        
        while index < len(numbers):
            j = len(numbers) - 1
            num = target - numbers[index]

            while j > index:
                if numbers[j] == num:
                    return [index + 1 , j + 1]
                else:
                    j -= 1
            
            index += 1

File: Data Structures & Algorithms/two-integer-sum/submission-2.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> indices; // Dictionary
        for(int i=0;i<nums.size();i++) {
            indices[nums[i]] = i;
        }
        for(int i=0;i<nums.size();i++){
            int diff=target-nums[i];
            if(indices.count(diff) && indices[diff]!=i){    // If the number is there & diff != num.
                return {i,indices[diff]};
            }    
        }
        return {};
    }
};
